File: framework/references/examples/node-backend/handler-example.ts
import { Request, Response } from 'express';
import { Result } from '../common/result';
import { UserRepository } from '../repositories/user.repository';
import { User } from '../domain/user';
import { logger } from '../common/logger';

interface CreateUserRequest {
  name: string;
  email: string;
}

interface CreateUserResponse {
  id: string;
  name: string;
  email: string;
}

export class CreateUserHandler {
  constructor(private readonly userRepo: UserRepository) {}

  async handle(req: Request): Promise<Result<CreateUserResponse>> {
    const { name, email } = req.body as CreateUserRequest;

    if (!name) return Result.fail('Name is required');
    if (!email) return Result.fail('Email is required');

    const user = User.create(name, email);
    const saved = await this.userRepo.save(user);
    if (saved.isFailure) return Result.fail(saved.error);

    logger.info('User created', { userId: saved.value.id });
    return Result.ok({
      id: saved.value.id,
      name: saved.value.name,
      email: saved.value.email,
    });
  }
}

export function createUserHandler(userRepo: UserRepository) {
  return async (req: Request, res: Response) => {
    const handler = new CreateUserHandler(userRepo);
    const result = await handler.handle(req);
    if (result.isSuccess) {
      res.status(201).json(result.value);
    } else {
      res.status(400).json({ error: result.error });
    }
  };
}

File: framework/references/examples/backend-dotnet/handler-example.cs
using YourProject.Domain;
using YourProject.Common;

namespace YourProject.Features.Users.Create;

public sealed record CreateUserCommand(string Name, string Email);

public sealed record CreateUserResponse(string Id, string Name, string Email);

public interface ICreateUserHandler
{
    Task<Result<CreateUserResponse>> Handle(CreateUserCommand command, CancellationToken ct);
}

public sealed class CreateUserHandler : ICreateUserHandler
{
    private readonly IUserRepository _userRepo;
    private readonly ILogger<CreateUserHandler> _logger;

    public CreateUserHandler(IUserRepository userRepo, ILogger<CreateUserHandler> logger)
    {
        _userRepo = userRepo;
        _logger = logger;
    }

    public async Task<Result<CreateUserResponse>> Handle(CreateUserCommand command, CancellationToken ct)
    {
        if (string.IsNullOrWhiteSpace(command.Name))
            return Result<CreateUserResponse>.Failure("Name is required");

        if (string.IsNullOrWhiteSpace(command.Email))
            return Result<CreateUserResponse>.Failure("Email is required");

        var userResult = User.Create(command.Name, command.Email);
        if (userResult.IsFailure)
            return Result<CreateUserResponse>.Failure(userResult.Error);

        var user = userResult.Value;
        var saveResult = await _userRepo.SaveAsync(user, ct);
        if (saveResult.IsFailure)
            return Result<CreateUserResponse>.Failure(saveResult.Error);

        _logger.LogInformation("User created: {UserId}", user.Id);

        return Result<CreateUserResponse>.Success(new CreateUserResponse(
            user.Id, user.Name, user.Email));
    }
}
